### PR TITLE
Set env vars in worker processes without going through `env` function.

### PR DESCRIPTION
**Pull request by rohinb2**

### CODE REVIEW

**Comment by rohinb2** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1274?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1273** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1273?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1274** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1274?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1272** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1272?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1271** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1271?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1152** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1152?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#1195](https://github.com/run-house/runhouse/pull/1195) <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1195?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#1270** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1270?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1269** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1269?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rohinb2 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: runhouse/resources/hardware/cluster.py [hunk 1]
```diff
@@ -1124,7 +1124,9 @@ def _start_or_restart_helper(
 
             env_vars = _process_env_vars(default_env.env_vars)
             if env_vars:
-                self.call(default_env.name, "_set_env_vars", env_vars)
+                self.set_process_env_vars(
```
Comment (jlewitt1) [inline runhouse/resources/hardware/cluster.py:1127]: does it still make sense to sync config env vars (ex: log level, observability) onto the cluster when syncing the default env, or should those be set as process env vars too?

Comment (rohinb2) [reply runhouse/resources/hardware/cluster.py:1127]: These need to be set in the default env -- better to set before the daemon starts. These env vars are set in the process

**Review by jlewitt1 — APPROVED** [review]

**Comment by rohinb2** [comment]
### Merge activity

* **Oct 15, 1:55 PM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/run-house/runhouse/1274).

**Pull request merged**